tests/unit/test_solvers/test_processed_variable_computed.py

```diff
@@ -65,6 +65,8 @@ def process_and_check_2D_variable(
     #    ::test_output_variables
     return y_sol, first_sol, second_sol, t_sol
 
+def call_func(t=None, x=None, r=None, y=None, z=None, R=None, warn=True, **kwargs):
+    return np.random.rand(5, 5)
```

Comment:
Could you improve the tests so they actually tests that the variables are passed correctly? At the moment it only checks that no error message is thrown.

Reply:
```
def test_call_default(self, mock_call_func):
        call_func()
        mock_call_func.assert_called()

    @patch('path.to.call_func')
    def test_call_with_t(self, mock_call_func):
        t_test = 0.5
        call_func(t=t_test)
        mock_call_func.assert_called_with(t=t_test)
.
.
.
.
```

Something like this? @brosaplanella

Reply:
I was thinking something more PyBaMM specific. Maybe something like this example (https://github.com/pybamm-team/PyBaMM/blob/369034fc92a4465ba9d27c290652b69ed0ae78c5/tests/unit/test_solvers/test_processed_variable.py#L139-L163), where a dummy model is created but instead of calling your space variable x, try calling it something that would be a kwarg above to see if it is passed correctly. Maybe call it `a` and try both passing `a` to see it works and passing `b` to check it throws an error.

Reply:
I think so. Note that you want `a` to be a `SpatialVariable` rather than an `InputParameter` because the latter needs to be passed at solving stage.